scripts/attribution_analysis.py

```diff
@@ -0,0 +1,63 @@
+import pandas as pd
+import seaborn as sb
+import scanpy as sc
+import matplotlib.pyplot as plt 
+import numpy as np
+
+
+class AttrAnalysis():
+    def __init__(self, data_path: str):
+        self.df = self.read_data(data_path)
+        
+    def read_data(self, data_path) -> pd.DataFrame:
+        df = pd.read_csv(data_path, index_col = 0)
+        genes = df.shape[0]
+        cell_types = df.shape[1]
+        print(f"The loaded data has {genes} genes and {cell_types} cell types")
+        return df
+    
+    
+    def get_attr_sign_count(self):
+        df = self.df
+        cell_type_names = df.columns.to_list()
+        attribution_count_df = pd.DataFrame(index=cell_type_names,columns=["negative","zero","postive"])
+        i = 0
+        for name in cell_type_names:
+            attribution_count_df.iloc[i,:] = [(df[name] < 0).sum(), (df[name] == 0).sum(), (df[name] > 0).sum()]
+            i+=1
+        return attribution_count_df
+    
+    def attr_sign_heat_map(self):
+        sb.heatmap(self.df, mask = (self.df>0) | (self.df<0), cbar = False, cmap = 'autumn')
```

Comment:
When calling plots within functions like this and elsewhere, it's nice to either return the `sb.heatmap` directly or assign `ax = sb.heatmap(...)` and return ax so the plot can be modified